### ISSUE TITLE
Feature: SLM-First Answer Architecture – Frontier Model Supervisor with Feedback Capture

### ISSUE THREAD

**Issue by voipin** [cloudwarriors-ai/cw-openclaw#17]
## Overview

Integrate the trained **SLM (from issue #5)** directly into OpenClaw's answer pipeline as the **first responder** for domain questions. A **frontier model (e.g., Claude, GPT-4o)** acts as a supervisor — evaluating the SLM's response, correcting it if needed, and delivering the final answer to the user. All interactions (SLM output, supervisor verdict, corrections) are saved for future training utilization.

## Architecture

```
User Question (in channel / DM)
        │
        ▼
   SLM Inference
   (domain-trained small model)
        │
        ▼
   SLM Draft Answer
        │
        ▼
 Frontier Model Supervisor
 (Claude / GPT-4o as judge)
        │
   ┌────┴────┐
   │         │
 ✅ Correct  ❌ Incorrect / Incomplete
   │         │
   │    Supervisor generates:
   │    - Corrected answer
   │    - Explanation of error
   │    - Guidance for SLM improvement
   │         │
   └────┬────┘
        │
        ▼
  Final Answer → User
        │
        ▼
  Save Interaction Record:
  - Original question
  - SLM draft answer
  - Supervisor verdict (correct/incorrect)
  - Corrected answer (if any)
  - Supervisor guidance/feedback
  - Timestamp, channel, user context
        │
        ▼
  Training Feedback Store (→ Issue #5 pipeline)
```

### SLM Integration
- [ ] Load and serve the domain-trained SLM checkpoint (from issue #5 training pipeline)
- [ ] Expose SLM inference as an internal API endpoint (e.g., `POST /slm/infer`)
- [ ] Route incoming domain questions through SLM first before frontier model
- [ ] Configurable confidence threshold — if SLM confidence is too low, skip to supervisor directly

### Frontier Model Supervisor
- [ ] Supervisor system prompt instructs it to:
  - Evaluate the SLM's answer for accuracy, completeness, and domain correctness
  - Return a structured verdict: `{ correct: true/false, corrected_answer: "...", guidance: "..." }`
  - Deliver the corrected/approved answer as the final user-facing response
- [ ] Supervisor feedback includes actionable guidance (not just "wrong" — explains *why*)
- [ ] Supervisor model is configurable (default: Claude Sonnet or GPT-4o)

### Interaction Logging & Feedback Capture
- [ ] Every SLM + supervisor interaction saved to feedback store:
  ```json
  {
    "id": "uuid",
    "timestamp": "2026-02-23T20:00:00Z",
    "channel": "zoom-support",
    "question": "How do I port a number to Zoom Phone?",
    "slm_answer": "Submit a port request in the Zoom admin portal under Phone > Port Numbers.",
    "supervisor_verdict": "correct",
    "corrected_answer": null,
    "supervisor_guidance": null,
    "final_answer": "Submit a port request in the Zoom admin portal under Phone > Port Numbers.",
    "slm_version": "v1.2.0",
    "supervisor_model": "claude-sonnet-4-6"
  }
  ```
- [ ] Incorrect verdicts automatically queued as labeled training data (→ Issue #5 feedback loop)
- [ ] Correct verdicts logged as positive reinforcement examples
- [ ] Guidance text stored as annotation for future fine-tuning

### OpenClaw Integration
- [ ] New answer pipeline mode: `slm-supervised` (toggleable per channel)
- [ ] Graceful fallback: if SLM inference fails or is unavailable, route directly to frontier model
- [ ] Latency guard: if SLM + supervisor round-trip exceeds threshold, return frontier answer directly
- [ ] Per-channel config to enable/disable SLM-first mode

### Observability
- [ ] Dashboard metrics:
  - SLM accuracy rate (% correct per day/week)
  - Questions routed through SLM vs. bypassed
  - Average latency: SLM-only vs. SLM+supervisor
  - Training data accumulation rate
- [ ] Admin ability to review recent SLM verdicts and override

## Supervisor System Prompt (Draft)

```
You are a supervisor evaluating answers generated by a small language model (SLM) trained on VoIP, UCaaS, and telephony support Q&A.

Given the user's question and the SLM's draft answer, evaluate:
1. Is the answer factually correct and complete?
2. Is it appropriate for the domain (Zoom Phone, Dialpad, RingCentral, UCaaS)?

Respond with a JSON object:
{
  "correct": true | false,
  "corrected_answer": "<improved answer or null if correct>",
  "guidance": "<brief explanation of what was wrong and how to improve, or null if correct>"
}

If the answer is correct, set corrected_answer and guidance to null.
Always provide the best possible final answer to the user.
```

## Integration Points

- **Issue #5 (SLM Training Pipeline)** — consumes trained model checkpoints; feeds back incorrect verdicts as new training data
- **Issue #4 (Stakeholders)** — notify ML team stakeholders when SLM accuracy drops below threshold
- **Issue #3 (DM Reminders)** — alert reviewers when feedback queue is large enough for a re-training run
- OpenClaw channel observe mode — SLM-supervised pipeline runs within existing observe/respond flow

## Notes

- This creates a self-improving system: the more questions OpenClaw answers, the more training data accumulates, the better the SLM gets
- Frontier model cost is only incurred for supervision — not full answer generation — reducing long-term API spend as SLM improves
- SLM should be evaluated on a held-out test set after each training round to verify improvement
- Consider A/B testing: route X% of traffic through SLM-supervised vs. frontier-only to measure quality delta

---
*Replicated from cloudwarriors-ai/moltbot-workspace#6*

**Comment by voipin** [comment]
Closed to match source status in cloudwarriors-ai/moltbot-workspace#6
